src/vhost_user/slave_req_handler.rs

```diff
@@ -645,23 +625,13 @@ impl<S: VhostUserSlaveReqHandler> SlaveReqHandler<S> {
         // instead of waiting for the call.
         let nofd = (msg.value & 0x100u64) == 0x100u64;
 
-        let mut rfd = None;
-        match rfds {
-            Some(fds) => {
-                if !nofd && fds.len() == 1 {
-                    rfd = Some(fds[0]);
-                } else if (nofd && !fds.is_empty()) || (!nofd && fds.len() != 1) {
-                    Endpoint::<MasterReq>::close_rfds(Some(fds));
-                    return Err(Error::InvalidMessage);
-                }
-            }
-            None => {
-                if !nofd {
-                    return Err(Error::InvalidMessage);
-                }
-            }
+        let file = take_single_file(files);
+
+        if (nofd && file.is_some()) || (!nofd && file.is_none()) {
```

Comment:
minor nit: I find double negatives really hard to read:  "doesn't not have an fd" vs "has an fd".

Reply:
declared `has_fd` instead.